.github/workflows/main.yml

```diff
@@ -1,26 +1,65 @@
-name: Docker Image CI
-
-on: [push]
+name: Build and  Publish Docker
+on: [push, pull_request]
 
 jobs:
-
   build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-16.04, self-hosted]
 
-    runs-on: ubuntu-latest
-    env:
-      DB_HOST: ${{ secrets.DB_HOST }}
-      DB_USER: ${{ secrets.DB_USER }}
-      DB_PASSWORD: ${{ secrets.DB_PASSWORD }}
-      DB_NAME: ${{ secrets.DB_NAME }}
-      DB_PORT: ${{ secrets.DB_PORT }}
-      PGADMIN_DEFAULT_EMAIL: ${{ secrets.PGADMIN_DEFAULT_EMAIL }}
-      PGADMIN_DEFAULT_PASSWORD: ${{ secrets.PGADMIN_DEFAULT_PASSWORD }}
-      AUTH_KEY: ${{ secrets.AUTH_KEY }}
-      SIGNIN_KEY: ${{ secrets.SIGNIN_KEY }}
+    services:
+      postgres:
+        # pulling docker hub
+        image: postgres:latest # same with the one which is written on docker-compose
+        env:
+          POSTGRES_PASSWORD: dummy-password
+          POSTGRES_USER: dummy-user
+          POSTGRES_DB: dummy-db
+            # Set health checks to wait until postgres has started
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+        ports:
+          # Maps tcp port 5432 on service container to the host
+          - 5432:5432
 
     steps:
-    - uses: actions/checkout@v1
-    - name: Build the docker-compose stack
-      run: docker-compose -f docker-compose.yml up -d
-    - name: Check running containers
-      run: docker ps -a
+      - name: Set up Go 1.14
+        uses: actions/setup-go@v1
+        with:
+          go-version: 1.14
+        id: go
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v1
+      - name: Get dependencies
+        run: |
+          go get -v -t -d ./...
+          if [ -f Gopkg.toml ]; then
+             curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
+             dep ensure
+          fi
+
+      - name: Test
+        run: |
+          go build -o server .
+          chmod +x ./server
+          ./server &
+          go test ./tests
+        env:  # this is nasty list of env variables, make it pretty
+          SSL_OFF: false  # which means ssl enabled
+          AUTH_KEY: random-key
+          CERT: ./tests/certs/localhost_50051.crt
+          CERT_KEY: ./tests/certs/localhost_50051.key
+          CA : ./tests/certs/haaukins-store.com.crt
+          SIGNIN_KEY: random-signin-key
+          DATABASE_HOST: localhost
+          DB_PORT: 5432
+          POSTGRES_USER: dummy-user
+          POSTGRES_PASSWORD: dummy-password
+          POSTGRES_DB: dummy-db
+          HOST: localhost:50051
```

Comment:
I do not think this is nice way of retrieving necessary information :( , there could be much better way of handling these parameters.

Reply:
yeah i agree... we can set up a configuration file for this repo as well as haaukins